src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -2,18 +2,70 @@
 
 import core.basesyntax.Storage;
 
+@SuppressWarnings("unchecked")
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private final K[] keys = (K[]) new Object[10];
+    private final V[] values = (V[]) new Object[10];
+    private int size;
+
+    public K[] getKeys() {
+        return keys;
+    }
+
+    public V[] getValues() {
+        return values;
+    }
+
+    public int getSize() {
+        return size;
+    }
+
     @Override
     public void put(K key, V value) {
+        int index = putValidator(getKeys(), getValues(), key);
+        if (index == -1) {
+            getKeys()[getSize()] = key;
+            getValues()[getSize()] = value;
+            grow();
+        } else if (index >= 0) {
+            getKeys()[index] = key;
+            getValues()[index] = value;
+        }
     }
 
     @Override
     public V get(K key) {
-        return null;
+        return getValidator(getKeys(), getValues(), key);
```

Comment:
just reuse getKeyIndex method insted
```suggestion
        int index = getKeyIndex(key);
        return index == -1 ? null : values[index];
```